src/odb/src/zutil/util.cpp

```diff
@@ -170,34 +150,84 @@ void RUDYCalculator::processMacroObstruction(odb::dbMaster* macro,
+                                              const int nets_per_tile)
+{
+  // Calculate the intersection range
+  const int minXIndex
+      = std::max(0, (obstruction_rect.xMin() - gridBlock_.xMin()) / tile_width);
+  const int maxXIndex
+      = std::min(tileCntX_ - 1,
+                 (obstruction_rect.xMax() - gridBlock_.xMin()) / tile_width);
+  const int minYIndex = std::max(
+      0, (obstruction_rect.yMin() - gridBlock_.yMin()) / tile_height);
+  const int maxYIndex
+      = std::min(tileCntY_ - 1,
+                 (obstruction_rect.yMax() - gridBlock_.yMin()) / tile_height);
+
+  // Iterate over the tiles in the calculated range
+  for (int x = minXIndex; x <= maxXIndex; ++x) {
+    for (int y = minYIndex; y <= maxYIndex; ++y) {
+      Tile& tile = getEditableTile(x, y);
+      const auto tileBox = tile.getRect();
+      if (obstruction_rect.overlaps(tileBox)) {
+        const auto hpwl = static_cast<float>(tileBox.dx() + tileBox.dy());
+        const auto wireArea = hpwl * wireWidth_;
+        const auto tileArea = tileBox.area();
+        const auto obstr_congestion = wireArea / tileArea;
+        const auto intersectArea = obstruction_rect.intersect(tileBox).area();
+
+        const auto tileObstrBoxRatio
+            = static_cast<float>(intersectArea) / static_cast<float>(tileArea);
+        const auto rudy
+            = obstr_congestion * tileObstrBoxRatio * 100 * nets_per_tile;
+        tile.addRUDY(rudy);
+      }
+    }
+  }
+}
 
-    // Calculate the intersection range
-    const int minXIndex
-        = std::max(0, (macroObstrRect.xMin() - gridBlock_.xMin()) / tileWidth);
-    const int maxXIndex = std::min(
-        tileCntX_ - 1, (macroObstrRect.xMax() - gridBlock_.xMin()) / tileWidth);
-    const int minYIndex
-        = std::max(0, (macroObstrRect.yMin() - gridBlock_.yMin()) / tileHeight);
-    const int maxYIndex
-        = std::min(tileCntY_ - 1,
-                   (macroObstrRect.yMax() - gridBlock_.yMin()) / tileHeight);
-
-    // Iterate over the tiles in the calculated range
-    for (int x = minXIndex; x <= maxXIndex; ++x) {
-      for (int y = minYIndex; y <= maxYIndex; ++y) {
-        Tile& tile = getEditableTile(x, y);
-        const auto tileBox = tile.getRect();
-        if (macroObstrRect.overlaps(tileBox)) {
-          const auto hpwl = static_cast<float>(tileBox.dx() + tileBox.dy());
-          const auto wireArea = hpwl * wireWidth_;
-          const auto obstr_congestion = wireArea / tileBox.area();
-          const auto intersectArea = macroObstrRect.intersect(tileBox).area();
-          const auto tileArea = tileBox.area();
-          const auto tileObstrBoxRatio = static_cast<float>(intersectArea)
-                                         / static_cast<float>(tileArea);
-          const auto rudy = obstr_congestion * tileObstrBoxRatio * 100 * 2;
-          tile.addRUDY(rudy);
-        }
+void RUDYCalculator::processTilesIntersectionSignalNet(const odb::Rect netBox,
```

Comment:
I would create a specific function for the supply nets instead of a function for the signal nets since the signal nets are handled according to the paper, and you added a specific way to support supply nets.

Reply:
The function for signal nets would not be strictly necessary, I just created it because I found it easier to read the code. You think I remove it either way?

Reply:
It is good to have the two separate functions; it helps readability a lot. I would rename the functions to make it clear that the "special case" is the supply nets and not the signal nets.

But this is just a minor comment, so feel free to choose what you think is best.